lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -342,6 +342,56 @@ static Inst *peerThroughFunctionConversions(SILValue value) {
   return nullptr;
 }
 
+/// Finds the differentiability witness corresponding to `attr` in `module`.
+static SILDifferentiabilityWitness *
+findDifferentiabilityWitness(SILModule &module, SILDifferentiableAttr *attr) {
+  auto *resultIndices =
+      IndexSubset::get(module.getASTContext(), 1, {attr->getIndices().source});
+  AutoDiffConfig config(attr->getIndices().parameters, resultIndices,
+                        attr->getDerivativeGenericSignature());
+  return module.lookUpDifferentiabilityWitness(
+      {attr->getOriginal()->getName(), config});
+}
+
+/// Sets the differentiability witness JVP and VJP to the JVP and VJP in `attr`.
+///
+/// `attr` must have a JVP and VJP.
+static void fillDifferentiabilityWitness(SILModule &module,
```

Comment:
Minor: how about settling on the verb "canonicalize" instead of "fill" and process" for `[differentiable]` attributes, differentiability witnesses, `differentiable_function` instructions?

`processDifferentiableAttribute` and `processDifferentiableFunctionInst` can also both be renamed.

Reply:
+1 (but only weakly)

Reply:
Sounds good. Done for this one.